gcloud-java-compute/src/main/java/com/google/cloud/compute/Compute.java

```diff
@@ -0,0 +1,2709 @@
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("field", field)
+          .add("operator", operator)
+          .add("value", value)
+          .toString();
+    }
+
+    String toPb() {
+      return field + ' ' + operator.name().toLowerCase() + ' ' + value.toString();
+    }
+  }
+
+  /**
+   * Class for filtering disk type lists.
+   */
+  class DiskTypeFilter extends ListFilter {
+
+    private static final long serialVersionUID = 4847837203592234453L;
+
+    private DiskTypeFilter(DiskTypeField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskTypeFilter equals(DiskTypeField field, String value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.EQ, checkNotNull(value));
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskTypeFilter notEquals(DiskTypeField field, String value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns an equals filter for the given field and long value.
+     */
+    public static DiskTypeFilter equals(DiskTypeField field, long value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.EQ, value);
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and long value.
+     */
+    public static DiskTypeFilter notEquals(DiskTypeField field, long value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.NE, value);
+    }
+  }
+
+  /**
+   * Class for filtering machine type lists.
+   */
+  class MachineTypeFilter extends ListFilter {
+
+    private static final long serialVersionUID = 7346062041571853235L;
+
+    private MachineTypeFilter(MachineTypeField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static MachineTypeFilter equals(MachineTypeField field, String value) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2740139423/issues/source?bid=3123104&fileBranchId=3354337#l608)